packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts

```diff
@@ -165,16 +165,29 @@ export function useGridColumns(columns: GridColumns, apiRef: GridApiRef): void {
     [forceUpdate, getColumnFromField, updateColumns],
   );
 
-  const moveColumn = React.useCallback(
+  const setColumnIndex = React.useCallback(
     (field: string, targetIndexPosition: number) => {
-      logger.debug(`Moving column ${field} to index ${targetIndexPosition}`);
       const oldIndexPosition = gridState.columns.all.findIndex((col) => col === field);
+      if (oldIndexPosition === targetIndexPosition) {
+        return;
+      }
+
+      logger.debug(`Moving column ${field} to index ${targetIndexPosition}`);
+
+      apiRef.current.publishEvent(GRID_COLUMN_ORDER_CHANGE, {
```

Comment:
small one. Could we type the params before publishing. So if we change the params then it breaks

Reply:
Also where is `options.onColumnOrderChange` hooked to the event?

Reply:
`useGridApiOptionHandler(apiRef, GRID_COLUMN_ORDER_CHANGE, options.onColumnOrderChange)` ?

Reply:
> Also where is options.onColumnOrderChange hooked to the event?

Do we need to call it with the event? Users can get it by subscribing to `GRID_COLUMN_REORDER_DRAG_END`.